test/plugins/auctionHouseModule/createBid.test.ts

```diff
@@ -0,0 +1,218 @@
+    auctionHouse: {
+      address: spokSamePubkey(auctionHouse.address),
+    },
+    mint: spok.notDefined,
+    token: {
+      address: findAssociatedTokenAccountPda(nft.mintAddress, seller.publicKey),
+      mint: {
+        address: spokSamePubkey(nft.mintAddress),
+      },
+    },
+    isPublic: false,
+  };
+  spok(t, bid, {
+    $topic: 'Bid',
+    ...expectedBid,
+  } as unknown as Specifications<Bid>);
+});
+
+test('[auctionHouseModule] create a new private bid by seller account on an Auction House', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const seller = await createWallet(mx);
+  const nft = await createNft(mx, {}, { owner: seller.publicKey });
+
+  const { auctionHouse, client } = await createAuctionHouse(mx);
+
+  // When we create a private bid on that NFT for 1 SOL.
+  const { bid, buyerTradeState } = await client
+    .bid({
+      mintAccount: nft.mintAddress,
+      seller: seller.publicKey,
+      price: sol(1),
+    })
+    .run();
+
+  // Then we created and returned the new Bid with appropriate defaults.
+  const expectedBid = {
+    tradeStateAddress: spokSamePubkey(buyerTradeState),
+    price: spokSameAmount(sol(1)),
+    tokens: spokSameAmount(token(1)),
+    auctionHouse: {
+      address: spokSamePubkey(auctionHouse.address),
+    },
+    mint: spok.notDefined,
+    token: {
+      address: findAssociatedTokenAccountPda(nft.mintAddress, seller.publicKey),
+      mint: {
+        address: spokSamePubkey(nft.mintAddress),
+      },
+    },
+    isPublic: false,
+  };
+  spok(t, bid, {
+    $topic: 'Bid',
+    ...expectedBid,
+  } as unknown as Specifications<Bid>);
+});
+
+test('[auctionHouseModule] create public receipt-less bid but cannot fetch it afterwards by default', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const seller = await createWallet(mx);
+  const nft = await createNft(mx, {}, { owner: seller.publicKey });
+
+  const { client } = await createAuctionHouse(mx);
+
+  // When we create a public bid on that NFT for 1 SOL.
+  const { bid, buyerTradeState } = await client
+    .bid({
+      mintAccount: nft.mintAddress,
+      seller: seller.publicKey,
+      price: sol(1),
+      printReceipt: false,
+    })
+    .run();
+
+  // Then we still get a bid model.
+  t.equal(bid.tradeStateAddress, buyerTradeState);
+  t.same(bid.price, sol(1));
+  t.same(bid.tokens, token(1));
+  t.ok(bid.isPublic);
+
+  // But we cannot retrieve it later with the default operation handler.
+  try {
+    await client.findBidByAddress(buyerTradeState).run();
+    t.fail('expected to throw AccountNotFoundError');
+  } catch (error: any) {
+    const hasNotFoundMessage = error.message.includes(
+      'The account of type [BidReceipt] was not found'
+    );
+    t.ok(error instanceof AccountNotFoundError, 'throws AccountNotFoundError');
+    t.ok(hasNotFoundMessage, 'has BidReceipt Not Found message');
+  }
```

Comment:
Could you simplify this using the helper method `assertThrows`?

```suggestion
  const promise = client.findBidByAddress(buyerTradeState).run();
  await assertThrows(t, promise, /Account Not Found/);
```

It's available in:

```ts
import { assertThrows } from '../../helpers';
```

Reply:
Yeah, I've just had it in a separate comment :)